cmd/otel-allocator/internal/collector/collector_test.go

```diff
@@ -157,6 +194,168 @@ func Test_runWatch(t *testing.T) {
 	}
 }
 
+func Test_gracePeriodWithNonRunningPodPhase(t *testing.T) {
+	namespace := "test-ns"
+	type args struct {
+		collectorMap map[string]*allocation.Collector
+	}
+	test := struct {
```

Comment:
I don't understand why this slice needs to exist. If there's only one test case, we can just run it directly, without test case structs or subtests.

Reply:
Sorry I am a bit lost here.... I think I've removed the slice? This is just a single construct?

Reply:
Ah, ok... I think I got you now. I can fix.

Reply:
Hey Mikolaj, since we are adding a `new Config: CollectorNotReadyGracePeriod` on TA, I will have more than 2 test cases per scenario. So I end up adding the test case struct and slice back!